CMakeLists.txt

```diff
@@ -1,155 +1,157 @@
-message( STATUS "Use this CMake flag to select build type: -DCMAKE_BUILD_TYPE=BUILD_TYPE")
-message( STATUS "Using build type: " ${CMAKE_BUILD_TYPE} )
-
 # Additonal compile flags for various compilers
 # ${CMAKE_CXX_COMPILER_ID} can be one of {GNU Clang AppleClang Intel MSVC}
 # - verbose output on loop vectorization
 if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang") # using Clang or AppleClang
-    #set(CMAKE_CXX_FLAGS
-    #    "${CMAKE_CXX_FLAGS} -Rpass-analysis=loop-vectorize -Rpass=loop-vectorize")
+  #set(CMAKE_CXX_FLAGS
+  #    "${CMAKE_CXX_FLAGS} -Rpass-analysis=loop-vectorize -Rpass=loop-vectorize")
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") # using GCC
-    # gcc 4.8/4.9 have for example no full regex implementation 
-    # and will cause runtime errors
-    # full c++11 support is only given in gcc 5.1
-    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5.1")
-        message(FATAL_ERROR "Version of gcc is too low, and does not have full c++11 support.")
-        message(FATAL_ERROR "Please install gcc 5.1 or higher")
-    endif()
-    #set(CMAKE_CXX_FLAGS
-    #    "${CMAKE_CXX_FLAGS} -ftree-vectorize -ftree-vectorizer-verbose=1")
+  # gcc 4.8/4.9 have for example no full regex implementation 
+  # and will cause runtime errors
+  # full c++11 support is only given in gcc 5.1
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5.1")
+    message(FATAL_ERROR "Version of gcc is too low, and does not have \
+    full c++11 support. Please install gcc 5.1 or higher.")
+  endif()
+  #set(CMAKE_CXX_FLAGS
+  #    "${CMAKE_CXX_FLAGS} -ftree-vectorize -ftree-vectorizer-verbose=1")
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel") # using Intel C++
-     # nothing here
+  # nothing here
 endif()
 
-# - Enable all warnings
+# Enable all warnings
 # This is not good code, in the sense that its not portable (across compilers)
 # The portable way using target_compile_options is just so inconvienient, that
 # we use this for now. 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
+list(APPEND CMAKE_CXX_FLAGS "-Wall")
 
 # Enable -fPIC flag
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
-message( STATUS "Using CXX compiler flags: " ${CMAKE_CXX_FLAGS} )
+message( STATUS "Global CXX compiler flags: " ${CMAKE_CXX_FLAGS} )
 
 # Setting name prefix for libraries
 set( CMAKE_SHARED_LIBRARY_PREFIX "libComPWA_" )
 
 #
 # External dependencies
 #
-
-# Frameworks for parallelization
-find_package( OpenMP QUIET )
-if (${OPENMP_CXX_FOUND})
-  message( STATUS "OpenMP version " ${OpenMP_CXX_VERSION} )
-endif()
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
 
 #
 # Boost unit_test_framework and serialization are used throughout the software
 # so that they are a requirement
 #
-find_package(Boost COMPONENTS
-	unit_test_framework
-    serialization
-    filesystem
-    system
-   	program_options
-    REQUIRED
+if(SAFE_LINKING)
+  # Link boost statically
+  set(Boost_USE_STATIC_LIBS ON)
+else()
+  add_definitions(-DBOOST_TEST_DYN_LINK=1)
```

Comment:
Don't add global definitions, especially that one.
See https://gist.github.com/mbinna/c61dbb39bca0e4fb7d1f73b0d66a4fd1